tests/counters.rs

```diff
@@ -0,0 +1,232 @@
+            .unwrap();
+
+        assert!(resp.status().is_success());
+        assert_eq!(counter_val.to_string(), resp.bytes().await.unwrap());
+    }
+
+    #[tokio::test]
+    #[serial]
+    async fn increment() {
+        cleanup().await;
+
+        let client = infinispan_client();
+        let counter_name = "test_counter";
+
+        client
+            .run(infinispan::request::counters::create_strong(counter_name))
+            .await
+            .unwrap();
+
+        client
+            .run(infinispan::request::counters::increment(counter_name))
+            .await
+            .unwrap();
+
+        let resp = client
+            .run(infinispan::request::counters::get(counter_name))
+            .await
+            .unwrap();
+
+        assert!(resp.status().is_success());
+        assert_eq!("1", resp.bytes().await.unwrap());
+    }
+
+    #[tokio::test]
+    #[serial]
+    async fn increment_with_delta() {
+        cleanup().await;
+
+        let client = infinispan_client();
+        let counter_name = "test_counter";
+
+        client
+            .run(infinispan::request::counters::create_strong(counter_name).with_value(1))
+            .await
+            .unwrap();
+
+        client
+            .run(infinispan::request::counters::increment(counter_name).by(2))
+            .await
+            .unwrap();
+
+        let resp = client
+            .run(infinispan::request::counters::get(counter_name))
+            .await
+            .unwrap();
+
+        assert!(resp.status().is_success());
+        assert_eq!("3", resp.bytes().await.unwrap());
+    }
+
+    #[tokio::test]
+    #[serial]
+    async fn delete() {
+        cleanup().await;
+
+        let client = infinispan_client();
+        let counter_name = "test_counter";
+
+        client
+            .run(infinispan::request::counters::create_weak(counter_name))
+            .await
+            .unwrap();
+
+        client
+            .run(infinispan::request::counters::delete(counter_name))
+            .await
+            .unwrap();
+
+        let resp = client
+            .run(infinispan::request::counters::get(counter_name))
+            .await
+            .unwrap();
+
+        assert_eq!(StatusCode::NOT_FOUND, resp.status());
+    }
+
+    #[tokio::test]
+    #[serial]
+    async fn list() {
+        cleanup().await;
+
+        let client = infinispan_client();
+        let counter_names: HashSet<String> =
+            HashSet::from_iter(vec!["counter_1".into(), "counter_2".into()]);
+
+        for counter_name in &counter_names {
+            client
+                .run(infinispan::request::counters::create_weak(counter_name))
+                .await
+                .unwrap();
+        }
+
+        let resp = client
+            .run(infinispan::request::counters::list())
+            .await
+            .unwrap();
+
+        assert_eq!(counter_names, counter_names_from_list_resp(resp).await);
+    }
+
+    fn infinispan_client() -> Infinispan {
```

Comment:
This should probably be somewhere in a common utils module for tests.

Reply:
Done :+1: